Embedded_Code/SSA/imu/imu.ino

```diff
@@ -0,0 +1,242 @@
+int irpin1 = A0; //3 analog IR sensors
+int irpin2 = A1;
+int irpin3 = A2;
+int sustrv = A3; //analog input for suspension travel, not implemented yet
+//int hallpin = P2.5; //Standard gpio for hall effect
+
+//read variables
+int irread1;
+int irread2;
+int irread3;
+int hallread;
+float voltage1, voltage2, voltage3;
+float temp1, temp2, temp3;
+
+//registers
+
+#define lsm9ds1_ag 0x6B //device reg
+
+#define gyro_control1 0x10 //Control of the gyro reg
+#define accel_control4 0x1E //Control of the accel reg4
+#define accel_control5 0x1F //Control of the accel reg5
+#define accel_control7 0x21 //Control of the accel reg5
+#define mag_pwr 0x16 //the register that powers the mangetometer
+//#define strt (0xD4>>1)
+#define strtw 0xD6 //send before write?
+#define strtr 0xD5
+
+#define gyroX0 0x18 //accel output registerss
+#define gyroX1 0x19
+#define gyroY0 0x1A
+#define gyroY1 0x1B
+#define gyroZ0 0x1C
+#define gyroZ1 0x1D
+
+#define accelX0 0x28 //accel output registerss
+#define accelX1 0x29
+#define accelY0 0x2A
+#define accelY1 0x2B
+#define accelZ0 0x2C
+#define accelZ1 0x2D
+
+void setup() {
+
+  pinMode(irpin1, INPUT);
+  pinMode(irpin2, INPUT);
+  pinMode(irpin3, INPUT);
+  pinMode(sustrv, INPUT);
+
+  I2C_init(); //initialize I2C transmission
+
+  Serial.begin(9600);
+}
+
+void loop() {
+
+//Note: delay is measured in milliseconds  
+
+//Reading IMU
+  getI2CData();
+  //delay(1000);
+  Serial.flush();
+}
+
+void I2C_init()
+{
+  pinMode(scl, INPUT_PULLUP);
+  pinMode(sda, INPUT_PULLUP);
+  
+  Wire.begin(); //initialize i2c transmission
+  Wire.beginTransmission(lsm9ds1_ag);
+  //Wire.write(strtw);
+  //init power modes
+  Wire.write(gyro_control1); //Control register for gyro
+  Wire.write(193); //Powers and set to 952 HZ, stock settings elsewhere, trial and error with filtering currently
+  Wire.write(accel_control4);
+  Wire.write(56); //Makes sure accel output is turned on
+  Wire.write(accel_control5);
+  Wire.write(56); 
+  Wire.write(accel_control7);
+  Wire.write(164);
+	
+  //Wire.write(mag_pwr); //Mag mode
+  Wire.endTransmission();
+}
+
+void getI2CData() //possible specify what data we want later instead of just including everything?
+{
+
+  int16_t rawaccelx, rawaccely, rawaccelz, rawgyrox, rawgyroy, rawgyroz;
+  float convaccelx, convaccely, convaccelz, convgyrox, convgyroy, convgyroz;
+  int8_t gX0, gX1, gY0, gY1, gZ0, gZ1, aX0, aX1, aY0, aY1, aZ0, aZ1;
+
+  Wire.beginTransmission(lsm9ds1_ag);
+  
+  Wire.write(gyroX0); //getting data from gyro registers
+  Wire.write(gyroX1);
+  Wire.write(gyroY0);
+  Wire.write(gyroY1);
+  Wire.write(gyroZ0);
+  Wire.write(gyroZ1);
+  
+  Wire.write(accelX0); //getting data from accelerometer registers
+  Wire.write(accelX1);
+  Wire.write(accelY0);
+  Wire.write(accelY1);
+  Wire.write(accelZ0);
+  Wire.write(accelZ1);
+
+  Wire.endTransmission(true); //continue transmission until reading is done
+
+  Wire.requestFrom(lsm9ds1_ag, 24); //requesting 12 bytes, or 3 16 bit numbers.
```

Comment:
for comment on this line: 12 bytes != 3,16bit numbers so i am confusion

Reply:
Old comment, will update